central/telemetry/billingmetrics/scheduler.go

```diff
@@ -0,0 +1,58 @@
+package billingmetrics
+
+import (
+	"time"
+
+	"github.com/stackrox/rox/central/sensor/service/pipeline/clustermetrics"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/logging"
+)
+
+const period = 1 * time.Hour
+
+var (
+	log  = logging.LoggerForModule()
+	stop = concurrency.NewSignal()
+)
+
+func gather() {
```

Comment:
Instead of having these as receiver-less functions, we should create a scheduler struct that has these functions as receiver

Reply:
I think an even better option is to move most of the logic to the respective datastore and use `gather()` to call that datastore. The scheduler struct will then only keep signal, ticker, and datastore reference.

Reply:
Refactored in #7137.